pybamm/solvers/base_solver.py

```diff
@@ -555,14 +562,23 @@ def solve(self, model, t_eval=None, external_variables=None, inputs=None):
             raise pybamm.SolverError("t_eval must increase monotonically")
 
         # Set up external variables and inputs
-        ext_and_inputs = self._set_up_ext_and_inputs(model, external_variables, inputs)
+        #
+        # Argument "inputs" can be either a list of input dicts or
+        # a single dict. The remaining of this function is only working
+        # with variable "input_list", which is a list of dictionaries.
+        # If "inputs" is a single dict, "inputs_list" is a list of only one dict.
+        inputs_list = inputs if isinstance(inputs, list) else [inputs]
+        ext_and_inputs_list = [
+            self._set_up_ext_and_inputs(model, external_variables, inputs)
+            for inputs in inputs_list
+        ]
 
         # Set up
         timer = pybamm.Timer()
 
         # Set up (if not done already)
         if model not in self.models_set_up:
-            self.set_up(model, ext_and_inputs, t_eval)
+            self.set_up(model, ext_and_inputs_list[0], t_eval)
```

Comment:
Assuming we models don't need to be set up for every value of the inputs, we just set it up using the first value.